### PR TITLE
feat: add the SLSA models for its v1.0 spec

### PR DESCRIPTION
**Pull request by chizhg**
BTW also fix the predicate type for SLSA v0.2 and related unit tests.

### CODE REVIEW

**Comment by chizhg** [comment]
@Alos could you have a look at this when you have time? Thanks!

**Comment by Alos** [comment]
Overall this looks good to me. @adityasaky would you like to approve this?

File: src/main/java/io/github/intoto/slsa/models/v02/Provenance.java [hunk 1]
```diff
@@ -109,6 +109,6 @@ public int hashCode() {
 
   @Override
   public String getPredicateType() {
-    return "https://slsa.dev/provenance/v0.1";
+    return "https://slsa.dev/provenance/v0.2";
```
Comment (adityasaky) [inline src/main/java/io/github/intoto/slsa/models/v02/Provenance.java:112]: v1?

Comment (SantiagoTorres) [reply src/main/java/io/github/intoto/slsa/models/v02/Provenance.java:112]: no I think this is ok. It's a fix on a previous issue

**Review by SantiagoTorres — APPROVED** [review]
this LGTM, though I think we are missing tests for v1?

File: src/main/java/io/github/intoto/slsa/models/v02/Provenance.java (hunk 1 above)
Comment (adityasaky) [reply src/main/java/io/github/intoto/slsa/models/v02/Provenance.java:112]: my bad!

**Comment by chizhg** [comment]
> this LGTM, though I think we are missing tests for v1?

I have created the unit tests, please take a second look. Thanks!

**Comment by chizhg** [comment]
Friendly ping on this. @SantiagoTorres @adityasaky could you take a second look and merge the PR if it looks good to you?

**Review by adityasaky — APPROVED** [review]
I think this is fine. @SantiagoTorres / @Alos could you confirm?

**Comment by Alos** [comment]
LGTM

**Pull request merged**